experimental/Schemes/Auxiliary.jl

```diff
@@ -0,0 +1,47 @@
+@Markdown.doc """
+    pullback(f::CoveredSchemeMorphism)
+
+Return a function `phi` which takes any reasonable 
+argument `v` associated to the `codomain` of `f` and 
+produces ``f*(v)``.
+"""
```

Comment:
This interface seems to be too generic to me.  Why return a function? 
I would argue for
`pullback(f::CoveredSchemeMorphism, I::IdealSheaf)`
`pullback(f::CoveredSchemeMorphism, F::CoherentSheaf)`
`pullback(f::CoveredSchemeMorphism, f::CoveredRationalFunction)`
which seems more transparent to me. Note that in some cases it is not really clear (at least to me) what the pullback is.

Reply:
Further I believe that anonymous functions make debugging a bit harder. 
Because in the traceback they only get some number and not a sensible name.

Reply:
> This interface seems to be too generic to me. Why return a function?

Why not? Then you can write `pbf = pullback(f)` and then later `pbf(I)` on your ideal sheaf. Just like you would have written it in a mathematics textbook. You get the $f^*$ object right in your hand. Don't you think this is nice?

Debugging: Yes, maybe it's a bit more tedious, but it works. 

Either way: If there are further objections to doing it this way, I mean also from the other Oscar-folks, then I can change this.

Reply:
In mathematics the $f^*$ is either a morphism and has a well defined domain and codomain 
or it is a functor. It has a domain category and a codomain category.
Now this thing is everything in one.  I can see that it is convenient and in math notation we just write f^* (but often also an additional sentence about it.)
But I am not so sure about the implications of this style of coding.

One drawback is that you cannot document its different dispatches separately. 
Any opinions @thofma  @fingolfin ?

Reply:
Handing around anonymous functions is not so nice for the reasons Simon mentions. Also it makes it rather awkward to write function signatures that accept such objects.

Another way would be returning some new struct and do the call overloading. Then one could document it and still have the `pbf(I)` syntax.

But I also agree with Simon that it is not clear what this actually is mathematically. I have never seen "let g be the pullback of f. Now consider g I" in a textbook (which would be how I read your code snippet with`pbf` above). Probably I should read different textbooks.

Reply:
I triggered some discussion on Slack about this today with various opinions. I think, this still has to develop a bit and we can change things, once we have a reliable outcome.

Reply:
Agreed.